src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -0,0 +1,113 @@
+
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles forum data (questions, comments, and replies) */
+@WebServlet("/account")
+public class UserDataServlet extends HttpServlet {
+  private static final String ID_PROPERTY = "id";
+  private static final String NAME_PROPERTY = "name";
+  private static final String EMAIL_PROPERTY = "email";
+  private static final String LIKED_PROPERTY = "liked";
+  private static final String COMMENT_PROPERTY = "comments";
+  private static final String IMAGE_PROPERTY = "image";
+
+  private static final String DATASTORE_USER = "User";
+
+  private String userId = "";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/plain");
+    response.getWriter().println(userId);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String idTokenString = request.getParameter("idtoken");
+    UrlFetchTransport transport = new UrlFetchTransport();
+
+    GsonFactory gsonFactory = new GsonFactory();
+
+    GoogleIdTokenVerifier verifier =
+        new GoogleIdTokenVerifier.Builder(UrlFetchTransport.getDefaultInstance(), gsonFactory)
+            .setAudience(
+                Collections.singletonList(
+                    "757099697912-i6jll98mfgochdo2vgjcovf64pepjesc.apps.googleusercontent.com"))
+            .build();
+
+    GoogleIdToken idToken;
+    try {
+      idToken = verifier.verify(idTokenString);
+    } catch (GeneralSecurityException e) {
+      idToken = null;
+    }
+    if (idToken != null) {
+      Payload payload = idToken.getPayload();
+
+      userId = payload.getSubject();
+      String email = payload.getEmail();
+      String name = (String) payload.get("name");
+      String pictureUrl = (String) payload.get("picture");
+      String locale = (String) payload.get("locale");
+      String familyName = (String) payload.get("family_name");
+      String givenName = (String) payload.get("given_name");
```

Comment:
Some of these fields look unused, and there's no corresponding reference to the field in `addUser()` below. What's the plan for these?

Reply:
Should we create a data object for this like we have for `ForumElement` and `Suggestion`?

https://github.com/googleinterns/granite2020/tree/master/src/main/java/com/google/sps/data

Reply:
Removed the unused fields. I also don't think its necessary because we're immediately passing it into the datastore, and we're not dealing with individual users outside of the one currently logged in.